src/document.jl

```diff
@@ -60,6 +60,9 @@ function get_offset(doc::Document, line::Integer, character::Integer)
     offset = line_offsets[line + 1]
     while character > 0
         offset = nextind(doc._content, offset)
+        if nextind(get_text(doc), offset) - offset > 2
```

Comment:
So this seems the wrong cut-off? UTF-8 uses more than 2 codepoints for anything above U+0800, but for UTF-16 the cutoff from 1 to 2 codepoints is 0x010000, I believe. So I think we need to capture the case where a character is above 0x010000, right? Also, wouldn't it be easier to directly check for the value of the char, like I did at https://github.com/julia-vscode/LanguageServer.jl/pull/419/files#diff-434665f0bad6ba3f5bf14654c456fa35R61?

@StefanKarpinski, can you help? I feel on very shaky ground whenever it comes to anything UTF :)

Reply:
Yes, I think 0x010000 is right.

Reply:
Gotcha, I've altered it to use an IOBuffer and explicitly check at those cutoffs